dspace-api/src/main/java/org/dspace/external/provider/impl/OrcidV2AuthorDataProvider.java

```diff
@@ -0,0 +1,280 @@
+            lastName = person.getName().getFamilyName().getValue();
+            externalDataObject.addMetadata(new MockMetadataValue("person", "familyName", null, null,
+                                                                 lastName));
+        }
+        if (person.getName().getGivenNames() != null) {
+            firstName = person.getName().getGivenNames().getValue();
+            externalDataObject.addMetadata(new MockMetadataValue("person", "givenName", null, null,
+                                                                 firstName));
+
+        }
+        externalDataObject.setId(person.getName().getPath());
+        externalDataObject
+            .addMetadata(new MockMetadataValue("dc", "identifier", "orcid", null, person.getName().getPath()));
+        externalDataObject
+            .addMetadata(new MockMetadataValue("dc", "identifier", "uri", null, orcidUrl + person.getName().getPath()));
+        if (!StringUtils.isBlank(lastName) && !StringUtils.isBlank(firstName)) {
+            externalDataObject.setDisplayValue(lastName + ", " + firstName);
+            externalDataObject.setValue(lastName + ", " + firstName);
+        } else if (StringUtils.isBlank(firstName)) {
+            externalDataObject.setDisplayValue(lastName);
+            externalDataObject.setValue(lastName);
+        } else if (StringUtils.isBlank(lastName)) {
+            externalDataObject.setDisplayValue(firstName);
+            externalDataObject.setValue(firstName);
+        }
+        return externalDataObject;
+    }
+
+    /**
+     * Retrieve a Person object based on a given orcid identifier
+     * @param id orcid identifier
+     * @return Person
+     */
+    public Person getBio(String id) {
+        log.debug("getBio called with ID=" + id);
+        if (!isValid(id)) {
+            return null;
+        }
+        InputStream bioDocument = orcidRestConnector.get(id + ((id.endsWith("/person")) ? "" : "/person"), accessToken);
+        XMLtoBio converter = new XMLtoBio();
+        Person person = converter.convertSinglePerson(bioDocument);
+        try {
+            bioDocument.close();
+        } catch (IOException e) {
+            log.error(e.getMessage(), e);
+        }
+        return person;
+    }
+
+    /**
+     * Check to see if the provided text has the correct ORCID syntax.
+     * Since only searching on ORCID id is allowed, this way, we filter out any queries that would return a
+     * blank result anyway
+     */
+    private boolean isValid(String text) {
+        return StringUtils.isNotBlank(text) && text.matches(ORCID_ID_SYNTAX);
+    }
+
+    @Override
+    public List<ExternalDataObject> searchExternalDataObjects(String query, int start, int limit) {
+        if (limit > 100) {
+            throw new IllegalArgumentException("The maximum number of results to retrieve cannot exceed 100.");
+        }
+
+        String searchPath = "search?q=" + URLEncoder.encode(query) + "&start=" + start + "&rows=" + limit;
```

Comment:
We have cases when the submitter knows the ORCID ID of the author. In those cases it's useful to ignore que ORCID search and simply do the record fetch. Otherwise, the query will return millions of results and we will do unnecessary requests, resulting in a delayed output for the client.

If we identify that the query value is an ORCID ID, then we firstly do simple checkdigit validation to ensure we have a valid ORCID ID. And if valid, then perform the record fetch.

How it works:
**ORCID-ID:** 0000-0003-1234-4321
**stripped(ORCID-ID):** 0000000312344321
**baseDigits:** 000000031234432 
**generateOrcidCheckDigit(baseDigits):** 1
**(stripped(ORCID-ID).equals(baseDigits+generateOrcidCheckDigit(baseDigits)))**: true


```

      /* Generates check digit as per ISO 7064 11,2.
       *
       */
       private static String generateOrcidCheckDigit(String baseDigits) {
             int total = 0;
             for (int i = 0; i < baseDigits.length(); i++) {
                    int digit = Character.getNumericValue(baseDigits.charAt(i));
                    total = (total + digit) * 2;
             }
             int remainder = total % 11;
             int result = (12 - remainder) % 11;
             return result == 10 ? "X" : String.valueOf(result);
       }
```

It can be considered in another PR as an improvement of Orcid Person provider.